transport/swift/swift.go

```diff
@@ -46,12 +47,25 @@ func (t transport) RoundTrip(req *http.Request) (resp *http.Response, err error)
 
 	object, objectHeaders, err := t.con.ObjectOpen(req.Context(), container, objectName, false, make(swift.Headers))
 
+	header := make(http.Header)
+
 	if err != nil {
+		if errors.Is(err, swift.ObjectNotFound) || errors.Is(err, swift.ContainerNotFound) {
+			return &http.Response{
+				StatusCode: http.StatusNotFound,
+				Proto:      "HTTP/1.0",
+				ProtoMajor: 1,
+				ProtoMinor: 0,
+				Header:     header,
+				Body:       nil,
```

Comment:
I'd add a body with the error message:

```suggestion
				Body:       io.NopCloser(strings.NewReader(err.Error())),
```

Reply:
Thanks for the quick review. It's updated now. Thanks @DarthSim